src/test/java/org/apache/maven/buildcache/its/junit/IntegrationTestExtension.java

```diff
@@ -208,6 +208,8 @@ public Object resolveParameter( ParameterContext parameterContext,
 
             Verifier verifier = new Verifier( testExecutionDir.toString(), true );
             verifier.setLogFileName( "../log.txt" );
+            verifier.setSystemProperty( "projectVersion", System.getProperty( "projectVersion" ) );
+            verifier.setLocalRepo( System.getProperty( "localRepo" ) );
```

Comment:
Please notice I also faced a similar problem [while adding tests](src/test/java/org/apache/maven/buildcache/its/ForkedExecutionsTest.java). But I decided to use temp directory this way to have clean state between runs at least for cache. 

So here the questions:
1. how this repository is cleaned?
2. Will it be unique between tests? Each integration test assumes clean state